sbi/inference/snpe/snpe_base.py

```diff
@@ -259,7 +284,7 @@ def _run_simulations(
         else:
             # XXX Make posterior.sample() accept tuples like prior.sample().
             theta = self._posterior.sample(
-                num_sims, x=self._x_o, show_progressbar=self._show_progressbar
+                num_sims, x=self._posterior.x_o, show_progressbar=self._show_progressbar
```

Comment:
There is no inherent reason why we should not **also** have it in `base`. But we definitely need it in posterior (e.g. as default when calling `sample()`). And since `base` has `posterior` as an attribute anyway I though we do not need to carry around `x_o` twice.